tests/marine/test_marine.py

```diff
@@ -2,85 +2,217 @@
+from marine import Marine
+
+
+# TODO: Add a test for FTP.
+
+
+@Parametrization.parameters(
+    "packet",
+    "fields_expected_fixture_name",
+    "extracted_fields_from_packet",
+    "bpf_filter",
+    "display_filter",
+)
+@Parametrization.case(
+    name="ARP packet",
+    packet=lazy_fixture("arp_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "broadcast_mac",
+        "mac_1",
+        "ip_1",
+        "broadcast_mac",
+        "ip_2",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_arp_packet"),
+    bpf_filter="arp",
+    display_filter="arp",
+)
+@Parametrization.case(
+    name="ICMP packet",
+    packet=lazy_fixture("icmp_packet"),
+    fields_expected_fixture_name=["mac_1", "mac_2", "ip_1", "ip_2", "icmp_echo_type"],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_icmp_packet"),
+    bpf_filter="ip",
+    display_filter="icmp",
+)
+@Parametrization.case(
+    name="TCP packet",
+    packet=lazy_fixture("tcp_packet"),
+    fields_expected_fixture_name=["mac_1", "mac_2", "ip_1", "ip_2", "port_1", "port_2"],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_tcp_packet"),
+    bpf_filter="ip",
+    display_filter="tcp",
+)
+@Parametrization.case(
+    name="DNS packet",
+    packet=lazy_fixture("dns_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "mac_2",
+        "ip_1",
+        "ip_2",
+        "port_3",
+        "dns_port",
+        "url_1",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_dns_packet"),
+    bpf_filter="ip",
+    display_filter="dns",
+)
+@Parametrization.case(
+    name="DHCP packet",
+    packet=lazy_fixture("dhcp_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "mac_2",
+        "ip_1",
+        "broadcast_ip",
+        "port_3",
+        "dhcp_port",
+        "ip_2",
+        "ip_1",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_dhcp_packet"),
+    bpf_filter="ip",
+    display_filter="dhcp",
+)
+@Parametrization.case(
+    name="HTTP packet",
+    packet=lazy_fixture("http_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "mac_2",
+        "ip_1",
+        "ip_2",
+        "port_1",
+        "http_port",
+        "http_type",
+        "http_uri",
+        "http_version",
+        "url_1",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_http_packet"),
+    bpf_filter="ip",
+    display_filter="http",
+)
+def test_packet_filter_and_parse(
+    request,
+    marine_instance: Marine,
+    packet: bytes,
+    fields_expected_fixture_name: List[Union[str, int]],
+    extracted_fields_from_packet: List[str],
+    bpf_filter: str,
+    display_filter: str,
+):
+    expected = dict(
+        zip(
+            extracted_fields_from_packet,
+            map(
+                lambda x: str(request.getfixturevalue(x)), fields_expected_fixture_name
```

Comment:
Do we always return strings? Does wireshark always return strings? Don't you define a type for every field in lua dissectors?

Reply:
Currently all of the results in the Python client are returned as strings, see this [open issue](https://github.com/tomlegkov/marine-python/issues/6).